scripts/local_env_create_vm

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env bash
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+
+## Shell Opts ----------------------------------------------------------------
+
+set -o pipefail
+set -xeuo
+
+if [ "$EUID" -ne 0 ]; then
+    echo "Local env needs sudo permissions"
+    exit 1
+fi
+
+SCRIPTPATH="$( cd "$(dirname "$0")" >/dev/null 2>&1 ; pwd -P )"
+
+CIFMW_LOCAL_ENV_BASE_DIR="${1}"
+CIFMW_LOCAL_ENV_BACKING_IMG_URL="${2}"
+CIFMW_LOCAL_ENV_PULL_SECRET_PATH="${3}"
```

Comment:
If a user runs like this 
```
CIFMW_LOCAL_ENV_BASE_DIR=${PWD} CIFMW_LOCAL_ENV_PULL_SECRET_PATH=~/pull-secret sudo make local_env_create
```
then it generally hit by following issue.
```
+ CIFMW_LOCAL_ENV_BASE_DIR=/opt/ci-framework
+ CIFMW_LOCAL_ENV_BACKING_IMG_URL=https://cloud.centos.org/centos/9-stream/x86_64/images/CentOS-Stream-GenericCloud-9-20230327.0.x86_64.qcow2
scripts/local_env_create_vm: line 32: 3: unbound variable
```
And if user runs like this
```
sudo CIFMW_LOCAL_ENV_BASE_DIR=${PWD} CIFMW_LOCAL_ENV_PULL_SECRET_PATH=/home/developer/pull-secret make local_env_create
```
It works fine. Can we handle the above issue also?

Reply:
Uhhhmm this is not related with the script. With the notation you propose you are passing to make the args as environment variables. As you are putting them before sudo, those values will be exported only in your current shell process and won't be passed to the child, and sudoed, shell unless you add -E to sudo.

Reply:
Agree with Pablo, also, this patch is lacking of a readme, it won't matter to have it in the Makefile if we don't have this information on the readme

Reply:
README and DOC (once we get the proper things set up, apparently using Sphinx and either gh-pages or readthedoc).